mods/mc_tutorial/init.lua

```diff
@@ -0,0 +1,1708 @@
+            local id = extract_id(id_string)
+            table.insert(context.tutorials.main.list, id_string)
+            mc_tutorial.record.temp[pname].dependents[id] = nil
+            mc_tutorial.record.temp[pname].depend_update.dep_nt[id] = false
+
+            context.tutorials.dep_nt.selected = math.max(1, math.min(context.tutorials.dep_nt.selected, #context.tutorials.dep_nt.list))
+            reload = true
+        end
+
+        -- MISC
+        if fields.finish then
+            if mc_tutorial.record.temp[pname].has_actions then
+                save_temp_fields(player, fields)
+
+                -- Build the tutorial table to send to mod storage
+                local recorded_tutorial = {
+                    dependencies = mc_tutorial.record.temp[pname].dependencies or {}, -- table of tutorial IDs that must be compeleted before the player can attempt this tutorial
+                    dependents = mc_tutorial.record.temp[pname].dependents or {}, -- table of tutorial IDs that completing this tutorial unlocks
+                    title = (mc_tutorial.record.temp[pname].title ~= "" and mc_tutorial.record.temp[pname].title) or "Untitled",
+                    length = mc_tutorial.record.temp[pname].sequence and #mc_tutorial.record.temp[pname].sequence or 0,
+                    seq_index = 1, -- default search always starts on the first element in the sequence
+                    description = (mc_tutorial.record.temp[pname].description ~= "" and mc_tutorial.record.temp[pname].description) or "No description provided",
+                    sequence = mc_tutorial.record.temp[pname].sequence or {},
+                    next_group = mc_tutorial.record.temp[pname].next_group or 1,
+                    on_completion = {
+                        message = (mc_tutorial.record.temp[pname].on_completion.message ~= "" and mc_tutorial.record.temp[pname].on_completion.message) or "You completed the tutorial!",
+                        items = mc_tutorial.record.temp[pname].on_completion.items or {},
+                        privs = mc_tutorial.record.temp[pname].on_completion.privs or {}
+                    },
+                    format = mc_tutorial.get_temp_shell().format
+                }
+
+                -- Send to mod storage
+                local tutorials = mc_tutorial.tutorials:to_table()
+                if not tutorials or not next(tutorials.fields) then
+                    mc_tutorial.tutorials:set_int("next_id", 2)
+                    mc_tutorial.tutorials:set_string("1", minetest.serialize(recorded_tutorial))
+                else
+                    local id = mc_tutorial.record.edit[pname] or mc_tutorial.tutorials:get("next_id") or 1
+                    
+                    -- Update dependencies/dependents in other tutorials
+                    for k,v in pairs(mc_tutorial.record.temp[pname].depend_update.dep_cy) do
```

Comment:
Hm, interesting. I thought I set things up so that the booleans would be in the values (`v`) and not the keys (`k`),  will check and see what's going on

Reply:
Was unable to replicate this, suspect that it might be caused by an issue with the ID extraction helper function or old format tutorials and storage conversion.

Reply:
Have added some additional checks that may prevent this from throwing an error, could you let me know what happens when you perform that sequence of actions now?
Additionally, could you let me know if you see ID numbers before the tutorials in the "Available tutorials" list in the "Dependencies" tab of the recording menu?